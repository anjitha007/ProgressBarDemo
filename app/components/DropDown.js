import React from 'react';
import PropTypes from 'prop-types';

const DropDown = (props) => {
  return (
    <div className="styled-select">
      <select onChange={props.onChange}>
        <option value="0">Progress#1</option>
        <option value="1">Progress#2</option>
        <option value="2">Progress#3</option>
      </select>
    </div>
  );
};

DropDown.propTypes = {
  onChange: PropTypes.func.isRequired,
};
export default DropDown;
