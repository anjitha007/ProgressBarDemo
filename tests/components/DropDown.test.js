import React from 'react';
import DropDown from '../../app/components/DropDown';

describe('DropDown Component', () => {
  let dropdownWrapper;
  const onChange = jest.fn();
  const options = [1, 2, 3];
  beforeEach(() => {
    dropdownWrapper = shallow(<DropDown onChange={onChange} options={options} />);
  });

  it('should exist', () => {
    expect(dropdownWrapper).toBeTruthy();
    expect(dropdownWrapper.find('.styled-select').type()).toEqual('div');
  });
  it('should display dropdown component with values', () => {
    expect(dropdownWrapper.find('select')).toBeTruthy();
    expect(dropdownWrapper.find('option').at(0).text()).toEqual("Progress#1");
    expect(dropdownWrapper.find('option').at(1).text()).toEqual("Progress#2");
    expect(dropdownWrapper.find('option').at(2).text()).toEqual("Progress#3");
  });

  it('should be able click the dropdown ', () => {
    expect(dropdownWrapper.find('select').length).toBe(1);
    expect(dropdownWrapper).toMatchSnapshot();
    const wrapper = shallow(<DropDown onChange={onChange} options={options} />);
    wrapper.find('select').simulate('change',  "40");
    expect(onChange).toBeCalledWith("40");
  });

});