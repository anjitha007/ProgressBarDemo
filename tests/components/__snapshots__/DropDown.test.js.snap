// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDown Component should be able click the dropdown  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DropDown
    onChange={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <select
        onChange={[MockFunction]}
      >
        <option
          value="0"
        >
          Progress#1
        </option>
        <option
          value="1"
        >
          Progress#2
        </option>
        <option
          value="2"
        >
          Progress#3
        </option>
      </select>,
      "className": "styled-select",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <option
            value="0"
          >
            Progress#1
          </option>,
          <option
            value="1"
          >
            Progress#2
          </option>,
          <option
            value="2"
          >
            Progress#3
          </option>,
        ],
        "onChange": [MockFunction],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Progress#1",
            "value": "0",
          },
          "ref": null,
          "rendered": "Progress#1",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Progress#2",
            "value": "1",
          },
          "ref": null,
          "rendered": "Progress#2",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Progress#3",
            "value": "2",
          },
          "ref": null,
          "rendered": "Progress#3",
          "type": "option",
        },
      ],
      "type": "select",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <select
          onChange={[MockFunction]}
        >
          <option
            value="0"
          >
            Progress#1
          </option>
          <option
            value="1"
          >
            Progress#2
          </option>
          <option
            value="2"
          >
            Progress#3
          </option>
        </select>,
        "className": "styled-select",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              value="0"
            >
              Progress#1
            </option>,
            <option
              value="1"
            >
              Progress#2
            </option>,
            <option
              value="2"
            >
              Progress#3
            </option>,
          ],
          "onChange": [MockFunction],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Progress#1",
              "value": "0",
            },
            "ref": null,
            "rendered": "Progress#1",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Progress#2",
              "value": "1",
            },
            "ref": null,
            "rendered": "Progress#2",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Progress#3",
              "value": "2",
            },
            "ref": null,
            "rendered": "Progress#3",
            "type": "option",
          },
        ],
        "type": "select",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
